File: src/test/java/com/raychenon/leetcode/ContainsDuplicateTest.java
package com.raychenon.leetcode;

import org.junit.BeforeClass;
import org.junit.Test;

import static junit.framework.TestCase.assertEquals;

/**
 * User: raychenon
 * Date: 24/3/2020
 * https://leetcode.com/problems/contains-duplicate/
 */
public class ContainsDuplicateTest {

    private static ContainsDuplicate instance;

    @BeforeClass
    public static void setUp() {
        instance = new ContainsDuplicate();
    }


    @Test
    public void containsTest() {
        int[] nums = {1, 2, 3, 1};
        assertWithVariousMethods(true, nums);
    }

    @Test
    public void noDuplicateTest() {
        int[] nums = {1, 2, 3, 4};
        assertWithVariousMethods(false, nums);
    }

    @Test
    public void longArrayTest() {
        int[] nums = {1, 1, 1, 3, 3, 4, 3, 2, 4, 2};
        assertWithVariousMethods(true, nums);
    }

    void assertWithVariousMethods(boolean expected, int[] array) {
        assertEquals(expected, instance.containsDuplicateSort(array));
        assertEquals(expected, instance.containsDuplicateSet(array));
    }


}

File: src/test/kotlin/com/raychenon/kotlin/leetcode/ContainsDuplicateTest.kt
package com.raychenon.kotlin.leetcode

import org.junit.Assert
import org.junit.Test

/**
 * User: raychenon
 * Date: 23/3/2020
 * https://leetcode.com/problems/contains-duplicate/
 */
class ContainsDuplicateTest {

    @Test
    fun containsTest() {
        assertWithVariousMethods(true, intArrayOf(1, 2, 3, 1))
    }

    @Test
    fun noDuplicateTest() {
        assertWithVariousMethods(false, intArrayOf(1, 2, 3, 4))
    }

    @Test
    fun longArrayTest() {
        assertWithVariousMethods(true, intArrayOf(1, 1, 1, 3, 3, 4, 3, 2, 4, 2))
    }


    fun assertWithVariousMethods(expected: Boolean, array: IntArray) {
        Assert.assertEquals(expected, ContainsDuplicate.containsDuplicateSort(array))
        Assert.assertEquals(expected, ContainsDuplicate.containsDuplicateSet(array))
    }

}